libbeat/docs/queueconfig.asciidoc

```diff
@@ -76,3 +77,147 @@ will be immediately available for consumption.
+spool are forwarded to the outputs, only after the write buffer has been
+flushed successfully.
+
+The spool waits for the output to acknowledge or drop events. If the spool is
+full, no new events can be inserted. The spool will block. Space is freed only
+after a signal from the output has been received.
+
+On disk, the spool operates in pages. The `file.page_size` setting configures
+the files page size on creation time. The optimal page size depends on the
+effective block size, used by the underlying file system.
+
+This sample configuration enables the spool with all default settings and the
+default file path:
+
+[source,yaml]
+------------------------------------------------------------------------------
+queue.spool: ~
+------------------------------------------------------------------------------
+
+This sample configuration creates a spool of 512MiB, with 16KiB pages. The
+write buffer is flushed if 10MiB of contents, or 1024 events have been
+written. If oldest available event is already waiting for 5s in the write
+buffer, the buffer will be flushed as well:
+
+[source,yaml]
+------------------------------------------------------------------------------
+queue.spool:
+  file:
+    path: "${path.data}/spool.dat"
+    size: 512MiB
+    page_size: 16KiB
+  write:
+    buffer_size: 10MiB
+    flush.timeout: 5s
+    flush.events: 1024
+------------------------------------------------------------------------------
+
+[float]
+==== Configuration options
+
+You can specify the following options in the `queue.spool` section of the
++{beatname_lc}.yml+ config file:
+
+[float]
+===== `file.path`
+
+The spool file path. The file is created on startup, if it does not exist.
+
+The default value is "${path.data}/spool.dat".
+
+===== `file.permissions`
+
+The file permissions. The permissions are applied when the file is
+created. In case the file already exists, The file permissions are compared
+with `file.permissions`. The spool file is not opened if the actual file
+permissions are more permissive then configured.
+
+The default value is 0600.
+
+
+===== `file.size`
+
+Spool file size. 
+
+The default value is 100 MiB.
+
+Note: The size should be much larger then the expected event sizes
+and write buffer size. Otherwise the queue will block, because it has not
+enough space.
+
+Note: The file size can not be changed once the file has been generated. This
+limitation will be removed in the future.
+
+
+===== `file.page_size`
+
+The files page size.
+
+The spool file is split into multiple pages of `page_size`. All write
+operations write a complete page.
+
+The default value is 4096 (4KiB).
+
+Note: This setting should match the file systems minimum block size. If the
+`page_size` is not a multiple of the file systems block size, the file system
+might create additional read operations on writes.
+
+Note: The page size is only set on file creation time. It can not be changed
```

Comment:
"only set at file creation time" sounds better to me. (comment applies to other instances)